File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        rightview = []
        deck = deque()
        if root:
            deck.append(root)
        while deck:
            rightSide = None
            for i in range(len(deck)):
                rightSide = deck.popleft()
                if rightSide.left:
                    deck.append(rightSide.left)
                if rightSide.right:
                    deck.append(rightSide.right)
            if rightSide:
                rightview.append(rightSide.val)
        return rightview

File: Data Structures & Algorithms/same-binary-tree/submission-3.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} p
     * @param {TreeNode} q
     * @return {boolean}
     */
    isSameTree(p, q) {
        // if both p and q are null then they are equal so return true
        if (!p && !q) return true
        // if either p is null or q is null or their values are not equal return false
        if (!(p && q) || p.val != q.val) return false
        // if left and right subtree are okay and the values are equal then return true else false
        return this.isSameTree(p.left, q.left) && this.isSameTree(p.right, q.right) && (p.val == q.val);
    }
}
